### PR TITLE
Add _findExecTraceAncestor() test for attacking the last branch with non-root-claim ancestor 

### PR DESCRIPTION
**Pull request by pochenai**
- Fix a bug of `_findExecTraceAncestor()` for attacking the last branch with non-root-claim ancestor 
- Add missing test for the above function, and pass `forge test --mt test_stepAttackDummyClaim_attack*`

### CODE REVIEW

File: packages/contracts-bedrock/src/dispute/FaultDisputeGameN.sol [hunk 1]
```diff
@@ -1100,7 +1093,7 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, ISemver {
         if (ancestorPos_.depth() == SPLIT_DEPTH + N_BITS) {
             ancestorClaim_ = ancestorClaimRoot;
         } else {
-            ancestorClaim_ = getClaim(ancestorClaimRoot.raw(), _pos, _daItem);
+            ancestorClaim_ = getClaim(ancestorClaimRoot.raw(), Position.wrap(_pos.raw() - 1), _daItem);
```
Comment (pochenai) [inline packages/contracts-bedrock/src/dispute/FaultDisputeGameN.sol]: Catch a bug here and fix it by subtracting 1 to get the right proof position

**Review by qizhou — COMMENTED** [review]
I would suggest putting cometic format change into another PR.

**Comment by pochenai** [comment]
> I would suggest putting cometic format change into another PR.

Yeah, fixed the commit.

**Review by qizhou — APPROVED** [review]

File: packages/contracts-bedrock/src/dispute/FaultDisputeGameN.sol (hunk 1 above)
Comment (pochenai) [reply packages/contracts-bedrock/src/dispute/FaultDisputeGameN.sol]: Actually, updated to `ancestorPos_`

**Pull request merged**